amazon-kinesis-client/src/main/java/software/amazon/kinesis/coordinator/logging/ExecutorStateEvent.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates.
+ * Licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package software.amazon.kinesis.coordinator.logging;
+
+import lombok.Getter;
+import lombok.ToString;
+
+@Getter
+@ToString
+public class ExecutorStateEvent extends DiagnosticEvent {
+    private int currentQueueSize;
+    private int activeThreads;
+    private int coreThreads;
+    private int leasesOwned;
+
+    protected ExecutorStateEvent(DiagnosticEventInput input) {
```

Comment:
Actually we can add more state information here,
getLargestPoolSize -> Returns the largest number of threads that have ever simultaneously been in the pool.
getMaximumPoolSize -> the maximum allowed number of threads

Reply:
Added new state information for getLargestPoolSize and getMaximumPoolSize